catalog/validations/plant-validations.js

```diff
@@ -3,32 +3,45 @@ const { body } = require('express-validator');
 const Plant = require('../models/plant');
 
 const plantValidationRules = () => {
-  const fieldToValidate = ['species', 'plantationType', 'plantationDate'];
-  return fieldToValidate.map(el => {
-    if (el === 'species') {
-      return body(el)
-        .trim()
-        .not()
-        .isEmpty()
-        .withMessage(`${el} cannot be empty.`)
-        .custom((value, { req }) => {
-          return Plant.where({ 
-              species: value,
-              variety: req.body.variety
-            })
-            .then(plant => {
-              if (plant.length > 0) {
-                return Promise.reject('A plant with this name/variety already exists.');
-              }
-            });
+  return [
+    body('species')
+    .trim()
+    .notEmpty()
+    .withMessage(`Species cannot be empty.`)
+    .custom((value, { req }) => {
+      return Plant.find({
+          species: value,
+          variety: req.body.variety
+        })
+        .then(plant => {
+          if (plant.length > 0 && req.method !== 'PUT') {
+            return Promise.reject('A plant with this name/variety already exists.');
+          }
+          if (plant.length > 0 && req.method === 'PUT' && plant[0]._id.toString() !== req.body._id.toString()) {
+            return Promise.reject('A plant with this name/variety already exists.');
```

Comment:
Why these to conditions are not combined into one?

Reply:
This validation is covering both creating a new plant (first condition => != 'PUT') and editing an existing plant (second condition == 'PUT') depending on the HTTP method used when this validation is ran.

Reply:
Yeah I get it but I think you can do instead something like:
``` javascript
if (plant.length > 0) {
  if (req.method === 'PUT' && plant[0]._id.toString() !== req.body._id.toString()) {
            return Promise.reject('Cannot update plant: a plant with this name/variety already exists.');
  } else {
            return Promise.reject('Cannot create plant: a plant with this name/variety already exists.');
}
```
This reduce a bit the repetition and avoid confusing message since you send the same string to the Promise.

Reply:
I couldn't come up with a better solution, but you're right, yours works a little bit better.